### PR TITLE
Dynamically create a firewall rule for test.

### LINKED ISSUE #94: Dynamically create a firewall rule for tests

### PR DESCRIPTION
**Pull request by tmatsuo**
fixes #94

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -44,3 +45,6 @@ install:
 
 script:
 - testing/run_test_suite.sh
+
+after_script:
```
Comment (bshaffer) [inline .travis.yml]: I think it'd be better to make the script name `test_cleanup.sh` or something along those lines, to match with the generic `install_test_dependencies.sh`. That way, if we have additional cleanup tasks/scripts, we can add them there.

**Comment by bshaffer** [comment]
one minor suggestion, otherwise LGTM

File: .travis.yml (hunk 1 above)
Comment (tmatsuo) [inline .travis.yml]: Makes sense, will do

**Comment by tmatsuo** [comment]
Thanks! I will merge once the test passes.

**Comment by tmatsuo** [comment]
Oops it hangs with:

```
gcloud compute firewall-rules delete ${FIREWALL_NAME} || /bin/true
The following firewalls will be deleted:
 - [travis-test]
Do you want to continue (Y/n)? 
```

I should add `CLOUDSDK_CORE_DISABLE_PROMPTS=1` envvar. I will update the PR.

**Comment by bshaffer** [comment]
good call

**Comment by tmatsuo** [comment]
The test failed. Very weird.

```
Created [https://www.googleapis.com/compute/v1/projects/getting-started-php/global/firewalls/travis-test].
NAME        NETWORK SRC_RANGES        RULES     SRC_TAGS TARGET_TAGS
travis-test default 104.197.119.59/32 tcp:27017          mongodb
```

```
There was 1 error:
1) Google\Cloud\Samples\Bookshelf\DataModel\MongoDbTest::testDataModel
MongoDB\Driver\Exception\RuntimeException: No suitable servers found (`serverselectiontryonce` set): [Failed connecting to '104.197.14.71:27017': Connection timed out]
```

```
$ testing/test_cleanup.sh
ERROR: (gcloud.compute.firewall-rules.delete) Some requests did not succeed:
 - The resource 'projects/getting-started-php/global/firewalls/travis-test' was not found
```

It is confusing, possible interpretation is that the creation somehow failed, but gcloud reported success.

**Comment by tmatsuo** [comment]
I found this

```
gcloud compute firewall-rules delete ${FIREWALL_NAME} || /bin/true
Deleted [https://www.googleapis.com/compute/v1/projects/getting-started-php/global/firewalls/travis-test].
```

Just before the creation call above. Another interpretation is that those 2 calls were conflicting each other, or executed in the reverse order. I added a sleep call after the deletion.

**Comment by tmatsuo** [comment]
Just a record, failure log is available at:
https://travis-ci.org/GoogleCloudPlatform/getting-started-php/jobs/109977272

**Comment by tmatsuo** [comment]
I came up with more possible hypothesis. It is possible that there were some stale build which execute the deletion code. I think this is true because even when there is no travis build running (at least travis UI said so), I saw a firewall `travis-test` existed on the developers console.

I will add timestamp to the firewall name.

**Comment by tmatsuo** [comment]
The hypothesis in https://github.com/GoogleCloudPlatform/getting-started-php/pull/95#issuecomment-185447635 seems correct. The firewall `travis-test` magically disappeared now.

**Comment by bshaffer** [comment]
hmm, very interesting. We have them only allowing one build at a time, so I wonder how this could have happened. In any case, nice work.

**Comment by tmatsuo** [comment]
Alright, merging

**Pull request merged**